src/inputs/internal/SelectFieldInput.tsx

```diff
@@ -208,10 +212,22 @@ export function SelectFieldInput<O, V extends Value>(props: SelectFieldInputProp
             setIsFocused(true);
           }}
           size={
+            // If sizeToContent, then, in order of precedence, base it of from:
+            // 1. input's value if any
+            // 2. If is MultiSelect and only one option is chosen, then use the length of that option to define the width to avoid size jumping on blur.
+            // 3. Use `nothingSelectedText`
+            // 4. Default to "1"
+            // And do not allow it to grow past a size of 20.
             sizeToContent
-              ? isFocused
-                ? Math.max(String(inputProps.value || "").length, 20)
-                : String(inputProps.value || "").length || 1
+              ? Math.min(
+                  String(
```

Comment:
Wrapping with `String()` specifically as `inputProps.value` can be ` string | number | readonly string[] | undefined`